geozero/src/wkb/wkb_reader.rs

```diff
@@ -26,38 +26,38 @@ impl<B: AsRef<[u8]>> GeozeroGeometry for Wkb<B> {
 )]
 #[cfg_attr(feature = "with-postgis-diesel", diesel(sql_type = Geometry))]
 #[cfg_attr(feature = "with-postgis-diesel", diesel(sql_type = Geography))]
-pub struct Ewkb(pub Vec<u8>);
```

Comment:
What do you think about a default trait implementation to make this change non-breaking (same for GpkgWkb, SpatiaLiteWkb, and MySQLWkb)?

Reply:
Oh I didn't realize that this was breaking, but I suppose it's breaking because the internal type is `pub`? So it still _accepts input_ with `Vec<u8>` but when you call `Ewkb(...).0` you no longer know you're getting a `Vec<u8>`?

Reply:
> default trait implementation

Do you have an example of what you mean?

Reply:
Like this: `pub struct Ewkb<B: AsRef<[u8]> = Vec<u8>>(pub B);`

Reply:
Interesting!

Reply:
```
error: defaults for type parameters are only allowed in `struct`, `enum`, `type`, or `trait` definitions
  --> geozero/src/wkb/wkb_reader.rs:29:17
   |
29 | pub struct Ewkb<B: AsRef<[u8]> = Vec<u8>>(pub B);
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
```

did I do that wrong?

Reply:
Hm. No, you did it right, but it looks like a bug in diesel's derive macros:

Here's an annotated snippet from cargo expand:

```
impl<'expr, B: AsRef<[u8]> = Vec<u8>> AsExpression<Geometry> for &'expr Ewkb<B> {
                          ^^^^^^^^^^- THIS IS A PROBLEM!!!
    type Expression = Bound<Geometry, Self>;
    fn as_expression(self) -> Self::Expression {
        Bound::new(self)
    }
}
```

I'll file a bug report with the diesel people. I guess we should just eat the breaking change?

Reply:
Filed: https://github.com/diesel-rs/diesel/issues/3902

Reply:
🤷‍♂️